### PR TITLE
Revising instructions to run back referencing scripts

### PR DESCRIPTION
**Pull request by amoghrajesh**
Revising the instructions on how to run the back referring scripts now that it is present with breeze

### CODE REVIEW

**Comment by amoghrajesh** [comment]
@potiuk looks to me that this is the final PR to wind up the issue :)

File: dev/README_RELEASE_AIRFLOW.md [hunk 1]
```diff
@@ -805,10 +805,7 @@ Documentation for providers can be found in the ``/docs/apache-airflow`` directo
 
     ```shell script
     breeze release-management publish-docs --package-filter apache-airflow --package-filter docker-stack
-    cd "${AIRFLOW_SITE_DIRECTORY}"
-    cd post-docs
-    python add-back-references.py airflow
-    cd ..
+    breeze release-management add-back-references --airflow-site-director DIRECTORY --gen-type airflow
```
Comment (potiuk) [inline dev/README_RELEASE_AIRFLOW.md]: You should not add the flag here. Those are real copy-pasteable instructions, it should **just-work** 

You should remove the flags, and ENVVAR will kick in here. If you look up AIRLFOW_SITE_DIRECTORY is set few steps above , so just running it without the flag should work here (and in the other places).

Comment (amoghrajesh) [reply dev/README_RELEASE_AIRFLOW.md]: Okay makes sense. Corrected it!

Comment (potiuk) [reply dev/README_RELEASE_AIRFLOW.md]: What I mean is to remove the flag altogether - this is how it works

**Review by potiuk — APPROVED** [review]

File: dev/README_RELEASE_PROVIDER_PACKAGES.md [hunk 2]
```diff
@@ -393,8 +393,7 @@ breeze release-management publish-docs \
     --override-versioned
 
 cd "${AIRFLOW_SITE_DIRECTORY}"
-cd post-docs
-python add-back-references.py providers
+breeze release-management add-back-references --airflow-site-directory --gen-type providers
```
Comment (eladkal) [inline dev/README_RELEASE_PROVIDER_PACKAGES.md:396]: this doesn't work for me

```
➜  airflow-site git:(main) ✗ breeze release-management add-back-references --airflow-site-directory --gen-type providers
                                                                                                                                                                                     
 Usage: breeze release-management add-back-references [OPTIONS]                                                                                                                      
                                                                                                                                                                                     
                                                                                                                                                                                     
 Try running the '--help' flag for more information.                                                                                                                                 
                                                                                                                                                                                     
╭─ Error ───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╮
│ Missing option '-g' / '--gen-type'.                                                                                                                                               │
╰───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╯
                                                                                                                                                                                     
                                                                                                                                                                                     
 To find out more, visit https://github.com/apache/airflow/blob/main/BREEZE.rst                                                                                                      
                                                                                                                                                                                     
    
```

Comment (amoghrajesh) [reply dev/README_RELEASE_PROVIDER_PACKAGES.md:396]: Thanks for reporting
I will have a look at it and fix it soon if there's an issue

Comment (amoghrajesh) [reply dev/README_RELEASE_PROVIDER_PACKAGES.md:396]: Can you try without the --airflow-directory? What does it complain?

Comment (potiuk) [reply dev/README_RELEASE_PROVIDER_PACKAGES.md:396]: Yes `--airflow-site-directory` is a clear problem. It treats `--gen-dir` as value of the `--airflow-site-directory`

Comment (amoghrajesh) [reply dev/README_RELEASE_PROVIDER_PACKAGES.md:396]: Yeah, i have fixed this in a follow up PR

**Pull request merged**